gradle-diff-task-dependencies/plugin/src/main/groovy/com/novoda/gradle/diffdependencytask/ConditionalDependency.groovy

```diff
@@ -0,0 +1,33 @@
+package com.novoda.gradle.diffdependencytask
+
+import org.gradle.api.Project
+import org.gradle.api.Task
+
+import java.util.regex.Pattern
+
+public class ConditionalDependency {
+
+    static {
+        Project.metaClass.registerConditionalDependency << { ConditionalDependency conditionalDependency ->
+            def config = delegate.extensions.getByName(Constants.TASK_CONFIG_PROPERTY_NAME) as DiffDependencyTaskConfig
+            config.conditionalDependencyRepository.add(conditionalDependency)
```

Comment:
But wouldn't this expose the container to project configuration? That's something I would like to avoid.